### ISSUE TITLE
Double call to server on app reload

### ISSUE THREAD

**Issue by vytautas-pranskunas-** [zino-app/graphql-flutter#627]
I have implemented everything as described here: https://github.com/zino-app/graphql-flutter/blob/master/examples/starwars/lib/episode/hero_query.dart

and added simple console.log('CCC') on my server GraphQL endpoint. After changing query variable value (hardcoded value for testing purposes) i see endpoint is hited 2 times. On refetch all works as expected.

![image](https://user-images.githubusercontent.com/3918999/81232209-c77ef580-8ff4-11ea-98ea-5f47b6c18907.png)

WOOOW - maybe I am doing something wrong but backend is called on every UI repaint!!! I am dragging slider and see backend service is called. 
Yes i do have cache enabled - i have tried with optimmistic cache and InMemory cache - same results

**Comment by micimize** [comment]
This is probably a configuration error. Few things –
* Why are there 3 ports listed
* Can you provide your client connection code in a `gist` or something
* Your client is maybe being instantiated inside of a re-rendering context
* This would probably be easier to troubleshoot in the `support` channel in [discord](https://discord.gg/tXTtBfC)

**Comment by vytautas-pranskunas-** [comment]
- 3 ports is because of local dev, emulator https dev and mobile https dev  - but i am using only emulators in this example
- I am taking everything litterally from link above
 
when i am calling client directly everything works as expected
```
var client = GraphQLProvider.of(context).value;
              var data = await client.query....
```

my code is:

main.dart
```
String get graphQLEndpoint {
  if (Platform.isAndroid) {
    return 'https://10.0.2.2:3000/graphql';
  } else {
    return 'https://localhost:3000/graphql';
  }
}

class DemoApp extends StatelessWidget {
    return ClientProvider(
        uri: graphQLEndpoint,
        child: MaterialApp(
          title: 'Flutter Demo',
          onGenerateRoute: RouteGenerator.generateRoute,
        ));
  }
}
```

client_provider.dart
```
import 'dart:io';
import 'package:http/io_client.dart';
import 'package:flutter/foundation.dart';

import 'package:graphql_flutter/graphql_flutter.dart';
import 'package:flutter/material.dart';

String uuidFromObject(Object object) {
  if (object is Map<String, Object>) {
    final String id = object['_id'].toString();
    if (id != "null" && id != null) {
      return id;
    }
  }
  return null;
}

final OptimisticCache cache = OptimisticCache(
  dataIdFromObject: uuidFromObject,
);

ValueNotifier<GraphQLClient> clientFor({
  @required String uri,
  String subscriptionUri,
}) {
  Link link;

  if (kReleaseMode) {
    link = HttpLink(uri: uri);
  } else {
    HttpClient httpClient = new HttpClient();
    httpClient.badCertificateCallback =
        (X509Certificate cert, String host, int port) => true;
    IOClient ioClient = new IOClient(httpClient);

    link = HttpLink(uri: uri, httpClient: ioClient);
  }

  if (subscriptionUri != null) {
    final WebSocketLink websocketLink = WebSocketLink(
      url: subscriptionUri,
      config: SocketClientConfig(
        autoReconnect: true,
        inactivityTimeout: Duration(seconds: 30),
      ),
    );

    /* final AuthLink authLink = AuthLink(
    getToken: () async => 'Bearer <YOUR_PERSONAL_ACCESS_TOKEN>',
    // OR
    // getToken: () => 'Bearer <YOUR_PERSONAL_ACCESS_TOKEN>',
  ); */

    link = link.concat(websocketLink);
  }

  return ValueNotifier<GraphQLClient>(
    GraphQLClient(
      cache: InMemoryCache(),
      link: link,
    ),
  );
}

/// Wraps the root application with the `graphql_flutter` client.
/// We use the cache for all state management.
class ClientProvider extends StatelessWidget {
  ClientProvider({
    @required this.child,
    @required String uri,
    String subscriptionUri,
  }) : client = clientFor(
          uri: uri,
          subscriptionUri: subscriptionUri,
        );

  final Widget child;
  final ValueNotifier<GraphQLClient> client;

  @override
  Widget build(BuildContext context) {
    return GraphQLProvider(
      client: client,
      child: child,
    );
  }
}


and my widget
```
class Test extends StatelessWidget {
  final testRequestParams request;
  const Test (this.request);

  @override
  Widget build(BuildContext context) {
    return Query(
      options: QueryOptions(
          documentNode: testQuery,
          variables: <String, dynamic>{
            'var2': this.request.var2,
            'var3': this.request.var3,
          }),
      builder: (
        QueryResult result, {
        Future<QueryResult> Function() refetch,
        FetchMore fetchMore,
      }) {
        print(result.data);
        if (result.hasException) {
          return Text(result.exception.toString());
        }

        if (result.loading) {
          return const Center(
            child: CircularProgressIndicator(),
          );
        }

        return Column(
          children: <Widget>[
            // Text(JsonEncoder.withIndent('  ').convert(result.data)),
            RaisedButton(
              onPressed: refetch,
              child: const Text('REFETCH'),
            ),
          ],
        );
      },
    );
  }
}
```
```

**Comment by nampdn** [comment]
@micimize I could confirm this issue has been fixed, when will it live in stable release?

**Comment by micimize** [comment]
@nampdn in the next one – I'm a bit swamped with other stuff but `beta` is stable enough to be merged/released right now

**Comment by micimize** [comment]
should be fixed in `v3.1.0` according to the above
